### ISSUE TITLE
session lost when opening an link from another application - follow up to 698

### ISSUE THREAD

**Issue by LaurenWags** [brave/brave-ios#1263]
### Description: 
Found while testing #698 

If Brave is closed (and not in memory), and you try to open a link from another application via long press > open in Brave, any tabs that you had open are gone. Brave opens with a single tab showing the link you selected, but all your previous tabs have vanished.

### Steps to Reproduce 
  1. Have several tabs open in Brave.
  2. Close Brave and remove from memory/background.
  3. Navigate to another app (twitter, slack, etc)
  4. Long press on a link.
  5. On the action sheet select Brave to open the link.

*Note, if you try this with Brave open in the background, this works as expected.

**Actual result:** 
Brave opens with a single tab, showing the link you selected. All your previous tabs are gone.

**Expected result:**
All your previous tabs should be retained and a new tab should open showing the link you selected.

**Reproduces how often:** [Easily reproduced, Intermittent Issue]
easily

**Brave Version:** 
1.10 (19.07.12.14)

**Device details:** 
iPad

**Website problems only:**
- did you check with Brave Shields down? 
- did you check in Safari/Firefox (WkWebView-based browsers)?  

### Additional Information 
cc @jhreis @danishjafri88

**Comment by LaurenWags** [comment]
Issue reproduces on latest Beta as well - 1.10 (19.07.16.21)

**Comment by kjozwiak** [comment]
Reproduced the issue on the `iPad Mini 4 running iOS 12.3.1` using `1.10 (19.07.16.21)`. Initially it looks like you've lost your entire session, but if close and restart Brave, the session will be restored and will include the link you've opened via a third party application as @LaurenWags mentioned above. Example:

* launch `1.10 (19.07.16.21)` and open ~2-3 tabs
* close brave and open another link in a third party application (Twitter) via Brave
* you'll notice that your session is missing all the tabs you've opened previously
* close and restart brave
* you'll notice that all your tabs are restored including the once you've opened using Twitter

@LaurenWags nice catch 👍 @jhreis probably something we want fixed for `1.10` as users might think they've lost their entire session.

**Comment by kjozwiak** [comment]
Looks like a recent regression, can't reproduce the above issue using `1.9.3 (19.05.22.15)` which is the latest released version.

**Comment by kylehickinson** [comment]
Regression caused by https://github.com/brave/brave-ios/pull/835

**Comment by danishjafri88** [comment]
Fixed in #1265

**Comment by LaurenWags** [comment]
* Verified passed on iPad 5th Gen (iOS 12.3.1) running 1.10 (19.07.16.23)
   - Verified STR from description
   - Verified using several apps (Twitter, Slack, etc) that when Brave was not running in the background/memory, and I attempted to open a link from selected app, Brave opened and retained all tabs from previous session. New tab was opened with the expected link in focus.
   - Verified with same apps that when Brave was running in the background/memory, and I attempted to open a link from the selected app, Brave opened and retained all tabs from previous session. New tab was opened with the expected link in focus.
* Verification **PASSED** on `iPhone 6s+ running iOS 12.3.1` using `1.10 (19.07.16.23)`
   - Verified STR from description
   - Verified using several apps (Twitter, Slack, etc) that when Brave was not running in the background/memory, and I attempted to open a link from selected app, Brave opened and retained all tabs from previous session. New tab was opened with the expected link in focus.
   - Verified with same apps that when Brave was running in the background/memory, and I attempted to open a link from the selected app, Brave opened and retained all tabs from previous session. New tab was opened with the expected link in focus.
